lib/linguist/heuristics.yml

```
@@ -594,6 +594,8 @@ disambiguations:
     pattern: '(?i:\bRebol\b)'
   - language: R
     pattern: '<-|^\s*#'
+  - language: Rez
+    pattern: '(#include\s+["<](Types\.r|Carbon/Carbon\.r)[">])|((resource|data|type)\s+\'[A-Za-z0-9]{4}\'\s+((\(.*\)\s+){0,1}){)'
```

You've got several problems here:

1. The `/` needs to be escaped
2. The `'` chars need to be escaped in the YAML way. No need for escaping in Ruby.

```suggestion
    pattern: '(#include\s+["<](Types\.r|Carbon\/Carbon\.r)[">])|((resource|data|type)\s+''[A-Za-z0-9]{4}''\s+((\(.*\)\s+){0,1}){)'
```